examples/VLCB4IN4OUT/VLCB4IN4OUT.ino

```diff
@@ -0,0 +1,550 @@
+  controller.begin();
+
+  Serial << F("> mode = ") << ((modconfig.currentMode) ? "Normal" : "Uninitialised") << F(", CANID = ") << modconfig.CANID;
+  Serial << F(", NN = ") << modconfig.nodeNum << endl;
+
+  // show code version and copyright notice
+  printConfig();
+
+  // set module parameters
+  VLCB::Parameters params(modconfig);
+  params.setVersion(VER_MAJ, VER_MIN, VER_BETA);
+  params.setModuleId(MODULE_ID);
+ 
+  // assign to Controller
+  controller.setParams(params.getParams());
+  controller.setName(mname);
+
+  // module reset - if switch is depressed at startup and module is in Uninitialised mode
+  if (userInterface.isButtonPressed() && modconfig.currentMode == VLCB::MODE_UNINITIALISED)
+  {
+    Serial << F("> switch was pressed at startup in Uninitialised mode") << endl;
+    modconfig.resetModule(&userInterface);
+  }
+
+  // opportunity to set default NVs after module reset
+  if (modconfig.isResetFlagSet())
+  {
+    Serial << F("> module has been reset") << endl;
+    modconfig.clearResetFlag();
+  }
+
+  // register our VLCB event handler, to receive event messages of learned events
+  ecService.setEventHandler(eventhandler);
+
+  // configure and start CAN bus and VLCB message processing
+  can2515.setNumBuffers(2, 1);      // more buffers = more memory used, fewer = less
+  can2515.setOscFreq(16000000UL);   // select the crystal frequency of the CAN module
+  can2515.setPins(10, 2);           // select pins for CAN bus CE and interrupt connections
+  if (!can2515.begin())
+  {
+    Serial << F("> error starting VLCB") << endl;
+  }
+}
+//
+///  setup Module - runs once at power on called from setup()
+//
+
+void setupModule()
+{
+  unsigned int nodeNum = modconfig.nodeNum;
+  // configure the module switches, active low
+  for (byte i = 0; i < NUM_SWITCHES; i++)
+  {
+    moduleSwitch[i].attach(SWITCH[i], INPUT_PULLUP);
+    moduleSwitch[i].interval(5);
+    switchState[i] = false;
+  }
+
+  // configure the module LEDs
+  for (byte i = 0; i < NUM_LEDS; i++) {
+    moduleLED[i].setPin(LED[i]);
+  }
+
+  Serial << "> Module has " << NUM_LEDS << " LEDs and " << NUM_SWITCHES << " switches." << endl;
+}
+
+
+void setup()
+{
+  Serial.begin(115200);
+  Serial << endl << F("> ** VLCB 4 in 4 out v1 ** ") << __FILE__ << endl;
+
+  setupVLCB();
+  setupModule();
+
+  // end of setup
+  DEBUG_PRINT(F("> ready"));
+}
+
+
+void loop()
+{
+  // do VLCB message, switch and LED processing
+  controller.process();
+
+  // process console commands
+  processSerialInput();
+
+  // Run the LED code
+  for (int i = 0; i < NUM_LEDS; i++) {
+    moduleLED[i].run();
+  }
+
+  // test for switch input
+  processSwitches();
+
+  //  processStartOfDay();
+}
+
+void processSwitches(void) {
+  bool isSuccess = true;
+  for (byte i = 0; i < NUM_SWITCHES; i++) {
+    moduleSwitch[i].update();
+    if (moduleSwitch[i].changed()) {      
+      byte nn_en[4];
+      modconfig.readEvent(i, nn_en);
+      byte evval = modconfig.getEventEVval(i, 1);
```

Comment:
Hmm, so here you use EV1 in the produced event. 
I think that in CANmINnOUT we used node variables for the switch behaviours, right?
Is it possible to set EV1 in these produced events from FCU?

Reply:
Yes.  It is as easy to change EV Values for produced events as it is consumed events in the FCU.  Just double click on them at the default window comes up.

Reply:
Produced events are now set as NVs and accessed through the application sketch.  This is to make life easier for COE (I think!)